test/jit/test_list_dict.py

```diff
@@ -143,6 +143,107 @@ def foo3():
         self.checkScript(foo3, ())
         FileCheck().check_count("aten::list", 2, exactly=True).run(torch.jit.script(foo3).graph)
 
+    def test_dict_keyword_with_kwargs(self):
+        def fn():
+            return dict(foo=1, bar=2, baz=3)
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_kwargs_using_container_values(self):
+        def fn():
+            return dict(foo=[1, 2, 3], bar=[4, 5, 6], baz=[7, 8, 9])
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_iterable(self):
+        def fn():
+            return dict([("foo", 1), ("bar", 2), ("baz", 3)])    # noqa: C406
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_empty_iterable(self):
+        def fn():
+            return dict([])
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_internal_aggregate_function(self):
+        def fn():
+            return dict(zip(["foo", "baz", "bar"], [1, 2, 3]))
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_mapping(self):
+        def fn():
+            return dict({"foo" : 1, "bar" : 2, "baz" : 3})
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_mapping_and_kwargs(self):
+        def fn():
+            return dict({"foo" : 1, "bar" : 2}, baz=3)
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_dict_comprehension(self):
+        def fn():
+            return dict({i: chr(i + 65) for i in range(4)})
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_dict_comprehension_and_kwargs(self):
+        def fn():
+            return dict({chr(65+i) : i for i in range(4)}, foo=2)
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_empty_dict_comprehension(self):
+        def fn():
+            return dict({})
+
+        self.checkScript(fn, ())
+
+    def test_dict_keyword_with_statically_unrolled_iterable(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.choices = torch.nn.ModuleDict({
+                        'conv': torch.nn.Conv2d(10, 10, 3),
+                        'pool': torch.nn.MaxPool2d(3)
+                })
+
+            def forward(self, x):
+                return dict(self.choices.items())
```

Comment:
hm, im not rly sure why there is working. `dict(zip([1, 2, 3], [1, 2, 3]))` might be an easier test case that exposes the same pathway (dict() of iterable)